docs/user/data-type.rst

```diff
@@ -0,0 +1,56 @@
+Data Types
```

Comment:
have 2 comments here:
1) you may refer some information from [here](https://docs.heavy.ai/sql/data-definition-ddl/datatypes-and-fixed-encoding#datatypes).
2) maybe we should don't put these detailed introduction as user manual? For user, what they need is more about how to build, install, configuration guide, trigger and may some easy debug guide.